src/rpc/mining.cpp

```diff
@@ -544,6 +544,7 @@ UniValue getblocktemplate(const JSONRPCRequest& request)
         pindexPrev = pindexPrevNew;
     }
     CBlock* pblock = &pblocktemplate->block; // pointer for convenience
+    assert(pblock);
```

Comment:
Ah, sorry. After re-reading the code I think I understand - your reading is that the `if`-branch will always be taken on the first call?